### ISSUE TITLE
Issue in place objects where we tab any place on the screen in ar-core android

### ISSUE THREAD

**Issue by ArigarasuthanRepo** [google-ar/arcore-android-sdk#144]
I have using ar-core sample project.i have placed object where we tab any place of the camera by using the code:
`                float sceneX=(tap.getX()/mSurfaceView.getMeasuredWidth())*2.0f-1.0f;
                float sceneY=(tap.getY()/mSurfaceView.getMeasuredHeight())*-2.0f+1.0f;

                Pose temp=frame.getCamera().getPose().compose(Pose.makeTranslation(sceneX,sceneY,-1.0f)).extractTranslation();

                Log.d("PoseTemp",""+temp);

                Log.d("CameraGetPose",""+camera.getPose());
                if (mAnchors.size() >= 1) {
                                mAnchors.get(0).detach();
                                mAnchors.remove(0);
                            }
                mAnchors.add(mSession.createAnchor(temp));`

   But objects were rotating while i was moving camera and placing anchor of the object.is it correct method to place object anywhere?

**Comment by Xanvial** [comment]
https://github.com/google-ar/arcore-android-sdk/pull/108

Is the sample code in there the same as what you want to do? Just change the translation vector to relative of your current touch

**Comment by ArigarasuthanRepo** [comment]
How to change translation vector relative to current touch?any examples help me to solve this issue

**Comment by ArigarasuthanRepo** [comment]
I will show an video demo for when i was change some location of the camera the object were rotated.
[obj rotation.zip](https://github.com/google-ar/arcore-android-sdk/files/1672510/obj.rotation.zip)

**Comment by Xanvial** [comment]
Is your program in the video already using code from #108 ?

```
private final Pose mCameraRelativePose = Pose.makeTranslation(0.15f, -0.15f, -0.5f);
```
Using above code create a translation that can be used to place an object almost center of the screen with depth about half meters. You can change the x value to your current x coordinate divided by display width, and so the y value. Note that I didn't compile this, so maybe there's some error

```
float x = touchX/displayWidth;   // save the touch value somewhere
float y = touchY/displayHeight;   // save the touch value somewhere
float z = -0.5f;                             // negative value will place it in front of the device, positive behind 
Pose tmpCameraRelativePose = Pose.makeTranslation(x, y, z);
frame.getCamera().getDisplayOrientedPose().compose(tmpCameraRelativePose).toMatrix(mAnchorMatrix, 0);
```

**Comment by ArigarasuthanRepo** [comment]
Thanks for the response. but i was placed object successfully my problem was objects were placed correctly at some positions only but other positions objects were rotated.object not same as one position to other positions that the issue i was mentioned above video.

**Comment by fredsa** [comment]
Translation isn't enough. You'll need to rotation as well. If you want the placed object to still face the camera when moved after the user rotates, you'll need to set the object rotation based on its position relative to the camera.

**Comment by ArigarasuthanRepo** [comment]
How to get the rotation when user move the camera and set the rotation?

**Comment by inio** [comment]
Look at the [rotateBetween function here](https://github.com/inio/arcore-android-sdk/blob/32e29839fb0ecf68e406c16344fc5be979c530e3/samples/hello_ar_java/app/src/main/java/org/inio/arcore/compassdemo/MathHelpers.java#L13-L35), it might help.

**Comment by inio** [comment]
Closing for inactivity.

**Comment by rtaha123** [comment]
hello, please can you help in how could we change the object already found in HelloAr activity and how do we modify the orientation of this object how to move?

**Comment by VidhiLatitude** [comment]
How can I put 3D object on onTouchevent and setScale for see far object ?
